app/src/org/commcare/models/database/SqlStorage.java

```diff
@@ -626,6 +633,7 @@ public void bulkRead(LinkedHashSet cuedCases, HashMap recordMap) {
                     c.moveToFirst();
                     int index = c.getColumnIndexOrThrow(DatabaseHelper.DATA_COL);
                     while (!c.isAfterLast()) {
+                        LifecycleSignaler.AssertNotAbandoned(abandonable);
```

Comment:
So just making sure I'm understanding-- this is the line that would stop the background thread for the badge calculations if its context has been marked as abandoned? If so, I'm definitely missing something. In the case that this throws an exception.. won't the program just crash? There's nothing in the code that catches/handles a `RequestAbandonedException`

Reply:
The new async layer that we are using handles errors for "Disposed" threads gracefully.

For right now this system basically isn't allowed to _guarantee_ that the exception that is thrown from this is a RequestAbandonedException, since the request abandon may end up clearing out memory which results in another error. Part of the semantics of an abandoned query are that once it's abandoned, it can throw any exceptions, since it's hard to know how it will fail.

Reply:
@ctsims But if this were called by anything other than the async framework that's handling the badge calculation, it would need real error handling, right?

Reply:
The reason I ask is because if so, I think this method declaration in `IStorageUtilityIndexed` should warn in some way that exceptions should be expected

Reply:
For sure. I'll update the docs to reflect that.